File: lc_p88_s1.py
from typing import *

class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        merged: List[int] = [0] * (m + n)
        i: int = 0
        j: int = 0
        k: int = 0

        while i < m and j < n:
            if nums1[i] <= nums2[j]:
                merged[k] = nums1[i]
                k += 1
                i += 1
            else:
                merged[k] = nums2[j]
                k += 1
                j += 1
        
        while i < m:
            merged[k] = nums1[i]
            k += 1
            i += 1

        while j < n:
            merged[k] = nums2[j]
            k += 1
            j += 1

        nums1[:] = merged


def main() -> None:
    nums1: List[int] = [1, 2, 3, 0, 0, 0]
    nums2: List[int] = [2, 5, 6]

    Solution().merge(nums1, 3, nums2, 3)
    print(nums1)


if __name__ == "__main__":
    main()

File: lc_p88_s1.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        std::vector<int> merged(m + n);
        int i = 0;
        int j = 0;
        int k = 0;

        while (i < m && j < n) {
            if (nums1[i] <= nums2[j]) {
                merged[k++] = nums1[i++];
            } else if (nums1[i] > nums2[j]) {
                merged[k++] = nums2[j++];
            }
        }

        while (i < m) {
            merged[k++] = nums1[i++];
        }

        while (j < n) {
            merged[k++] = nums2[j++];
        }

        nums1 = merged;
    }
};

int main(void) {
    std::vector<int> nums1 = {1, 2, 3, 0, 0, 0};
    std::vector<int> nums2 = {2, 5, 6};
    Solution().merge(nums1, 3, nums2, 3);
    
    for (const auto& i : nums1) {
        std::cout << i << " ";
    }
    return 0;
}
